From 05cd424490974a6d55f1b827d7a18520661da4b8 Mon Sep 17 00:00:00 2001
From: wanghang73 <wanghang73@huawei.com>
Date: Tue, 21 Apr 2026 11:46:32 +0800
Subject: [PATCH 16/48] urma: remove bonding device unused code

(cherry picked from commit 6ef9468ce5ad7bbaf20db3f0a87d65adc39cedac)
---
 src/urma/lib/urma/bond/bondp_api.c       |  18 +---
 src/urma/lib/urma/bond/bondp_datapath.c  |   4 -
 src/urma/lib/urma/bond/bondp_types.c     |  25 -----
 src/urma/lib/urma/bond/bondp_types.h     |  26 +----
 src/urma/lib/urma/bond/utils/bdp_queue.c | 118 -----------------------
 src/urma/lib/urma/bond/utils/bdp_queue.h |  61 ------------
 6 files changed, 9 insertions(+), 243 deletions(-)
 delete mode 100644 src/urma/lib/urma/bond/bondp_types.c
 delete mode 100644 src/urma/lib/urma/bond/utils/bdp_queue.c
 delete mode 100644 src/urma/lib/urma/bond/utils/bdp_queue.h

diff --git a/src/urma/lib/urma/bond/bondp_api.c b/src/urma/lib/urma/bond/bondp_api.c
index a5e3c5e7..9ed46586 100644
--- a/src/urma/lib/urma/bond/bondp_api.c
+++ b/src/urma/lib/urma/bond/bondp_api.c
@@ -141,10 +141,6 @@ static int bondp_delete_vjfce(bondp_jfce_t *bdp_jfce)
 urma_jfce_t *bondp_create_jfce(urma_context_t *ctx)
 {
     bondp_context_t *bdp_ctx = CONTAINER_OF_FIELD(ctx, bondp_context_t, v_ctx);
-    if (!is_valid_ctx(bdp_ctx)) {
-        URMA_LOG_ERR("Invalid bond ctx.\n");
-        return NULL;
-    }
 
     bondp_jfce_t *bdp_jfce = (bondp_jfce_t *)calloc(1, sizeof(bondp_jfce_t));
     if (bdp_jfce == NULL) {
@@ -378,6 +374,11 @@ urma_status_t bondp_delete_jfc(urma_jfc_t *jfc)
     return ret;
 }
 
+static inline int get_matrix_port_p_idx(int primary_idx, int port_idx)
+{
+    return primary_idx * PORT_EID_MAX_NUM_PER_DEV + port_idx + PRIMARY_EID_NUM;
+}
+
 static int convert_bond_port_id_to_active_index(const bondp_context_t *bdp_ctx, bondp_port_id_t port_id,
                                                 uint32_t *active_index)
 {
@@ -1564,11 +1565,6 @@ urma_status_t bondp_unimport_jetty(urma_target_jetty_t *target_jetty)
     bondp_context_t *bdp_ctx = CONTAINER_OF_FIELD(target_jetty->urma_ctx, bondp_context_t, v_ctx);
     urma_status_t ret = URMA_SUCCESS;
 
-    if (!is_valid_bdp_tjetty(bdp_tjetty)) {
-        URMA_LOG_ERR("Invalid bdp tjetty\n");
-        return URMA_EINVAL;
-    }
-
     bondp_unregister_health_check_task(bdp_ctx, bdp_tjetty);
 
     if (bondp_unimport_pjetty(bdp_tjetty) != URMA_SUCCESS) {
@@ -1771,10 +1767,6 @@ urma_status_t bondp_unimport_jfr(urma_target_jetty_t *target_jfr)
     bondp_target_jetty_t *bdp_tjetty = CONTAINER_OF_FIELD(target_jfr, bondp_target_jetty_t, v_tjetty);
     urma_status_t ret = URMA_SUCCESS;
 
-    if (!is_valid_bdp_tjetty(bdp_tjetty)) {
-        URMA_LOG_ERR("Invalid bdp tjetty\n");
-        return URMA_EINVAL;
-    }
     if (bondp_unimport_pjfr(bdp_tjetty) != URMA_SUCCESS) {
         ret = URMA_FAIL;
     }
diff --git a/src/urma/lib/urma/bond/bondp_datapath.c b/src/urma/lib/urma/bond/bondp_datapath.c
index df279a05..382fb35f 100644
--- a/src/urma/lib/urma/bond/bondp_datapath.c
+++ b/src/urma/lib/urma/bond/bondp_datapath.c
@@ -156,10 +156,6 @@ static urma_status_t post_send_check_valid(bondp_comp_t *bdp_send_comp, bondp_ta
         URMA_LOG_ERR("Try to call post_send api by invalid comp_type: %d\n", bdp_send_comp->comp_type);
         return URMA_EINVAL;
     }
-    if (!is_valid_bdp_tjetty(bdp_tjetty)) {
-        URMA_LOG_ERR("Invalid bdp_target_jetty\n");
-        return URMA_EINVAL;
-    }
     urma_status_t ret = post_send_check_jfs_wr_valid(bdp_send_comp->bondp_ctx, wr);
     if (ret != URMA_SUCCESS) {
         return ret;
diff --git a/src/urma/lib/urma/bond/bondp_types.c b/src/urma/lib/urma/bond/bondp_types.c
deleted file mode 100644
index 1e00ea20..00000000
--- a/src/urma/lib/urma/bond/bondp_types.c
+++ /dev/null
@@ -1,25 +0,0 @@
-/*
- * SPDX-License-Identifier: MIT
- * Copyright (c) Huawei Technologies Co., Ltd. 2025-2025. All rights reserved.
- * Description: Implementation of bonding provider support functions in bondp_types
- * Author: Ma Chuan
- * Create: 2025-02-12
- * Note:
- * History: 2025-02-12  Create file
- */
-#include "bondp_types.h"
-
-bool is_valid_ctx(bondp_context_t *ctx)
-{
-    return ctx && is_valid_dev_num(ctx->dev_num);
-}
-
-bool is_valid_bdp_tjetty(bondp_target_jetty_t *bdp_tjetty)
-{
-    return bdp_tjetty;
-}
-
-bool is_valid_import_tseg(bondp_import_tseg_t *rtseg)
-{
-    return rtseg;
-}
diff --git a/src/urma/lib/urma/bond/bondp_types.h b/src/urma/lib/urma/bond/bondp_types.h
index 6a119aff..092f9bcb 100644
--- a/src/urma/lib/urma/bond/bondp_types.h
+++ b/src/urma/lib/urma/bond/bondp_types.h
@@ -7,26 +7,25 @@
  * Note:
  * History: 2025-02-05   Create File
  */
+
 #ifndef BONDP_TYPES_H
 #define BONDP_TYPES_H
 
-#include <pthread.h>
 #ifndef __cplusplus
 #include <stdatomic.h>
 #else
 #include <atomic>
 #endif
+#include <stdbool.h>
+
 #include "bondp_hash_table.h"
 #include "bondp_wr_buf.h"
 #include "topo_info.h"
 #include "ub_list.h"
 #include "urma_types.h"
 #include "urma_ubagg.h"
-#include <stdbool.h>
 
-#define BONDP_MAX_NUM_JFRS            (10240)
 #define BONDP_MAX_NUM_JETTYS          (10240)
-#define BONDP_MAX_NUM_SEGS            (10240)
 #define BONDP_MAX_NUM_RSEGS           (10240)
 #define BONDP_MAX_WR_LIST_NUM         (300)
 #define BONDP_MAX_SGE_NUM             (32)
@@ -44,7 +43,6 @@
 
 struct bondp_target_jetty;
 
-
 typedef enum bondp_health_mode {
     HEALTH_MODE_BACKUP_CHECK,
     HEALTH_MODE_PRIMARY_CHECK,
@@ -285,23 +283,6 @@ typedef struct urma_bond_id_info_out {
     bool connected[URMA_UBAGG_DEV_MAX_NUM][URMA_UBAGG_DEV_MAX_NUM];
 } urma_bond_id_info_out_t;
 
-static inline bool is_valid_dev_num(int dev_num)
-{
-    return dev_num > 0 && dev_num <= URMA_UBAGG_DEV_MAX_NUM;
-}
-
-bool is_valid_ctx(bondp_context_t *ctx);
-
-bool is_valid_bdp_tjetty(bondp_target_jetty_t *bdp_tjetty);
-
-bool is_valid_import_tseg(bondp_import_tseg_t *rtseg);
-
-/* Get index of matrix server port in p_ctx, p_jetty, etc. */
-static inline int get_matrix_port_p_idx(int primary_idx, int port_idx)
-{
-    return primary_idx * PORT_EID_MAX_NUM_PER_DEV + port_idx + PRIMARY_EID_NUM;
-}
-
 static inline bool is_empty_eid(urma_eid_t *eid)
 {
     return eid->in6.interface_id == 0 && eid->in6.subnet_prefix == 0;
@@ -311,4 +292,5 @@ static inline bool is_single_dev_mode(bondp_context_t *ctx)
 {
     return ctx->bonding_mode == BONDP_BONDING_MODE_STANDALONE;
 }
+
 #endif // BONDP_TYPES_H
diff --git a/src/urma/lib/urma/bond/utils/bdp_queue.c b/src/urma/lib/urma/bond/utils/bdp_queue.c
deleted file mode 100644
index cc199fbb..00000000
--- a/src/urma/lib/urma/bond/utils/bdp_queue.c
+++ /dev/null
@@ -1,118 +0,0 @@
-/*
- * SPDX-License-Identifier: MIT
- * Copyright (c) Huawei Technologies Co., Ltd. 2025-2025. All rights reserved.
- * Description: bondp queue implementation
- * Author: Ma Chuan
- * Create: 2025-03-05
- * Note:
- * History:
- */
-#include <stdlib.h>
-#include "urma_log.h"
-#include "bdp_queue.h"
-
-void bdp_queue_init(bdp_queue_t *q, uint32_t max_node)
-{
-    STAILQ_INIT(&q->head);  /* Initialize the queue. */
-    q->node_num = 0;
-    q->max_node = max_node;
-}
-
-void bdp_queue_uninit(bdp_queue_t *q)
-{
-    bdp_queue_node_t *cur = NULL;
-
-    while (STAILQ_EMPTY(&q->head) == false) {
-        cur = STAILQ_FIRST(&q->head);
-        STAILQ_REMOVE_HEAD(&q->head, nodes);
-        free(cur);
-    }
-}
-
-int bdp_queue_front(bdp_queue_t *q, void **data)
-{
-    if (data == NULL) {
-        URMA_LOG_ERR("data is NULL\n");
-        return -1;
-    }
-    if (STAILQ_EMPTY(&q->head)) {
-        return -1;
-    }
-    *data = STAILQ_FIRST(&q->head)->data;
-    return 0;
-}
-
-int bdp_queue_push_tail(bdp_queue_t *q, void *data)
-{
-    if (q->node_num >= q->max_node) {
-        URMA_LOG_ERR("Failed to enqueue with invalid node_num: %u, max_node: %u.\n", q->node_num, q->max_node);
-        return -1;
-    }
-
-    bdp_queue_node_t *node = NULL;
-
-    node = malloc(sizeof(bdp_queue_node_t));
-    if (node == NULL) {
-        URMA_LOG_ERR("Failed to alloc bdp_queue_node\n");
-        return BDP_QUEUE_ALLOC_ERR;
-    }
-    node->data = data;
-    q->node_num++;
-    STAILQ_INSERT_TAIL(&q->head, node, nodes);
-    return 0;
-}
-
-int bdp_queue_pop_head(bdp_queue_t *q, void **data)
-{
-    if (STAILQ_EMPTY(&q->head)) {
-        return -1;
-    }
-    if (data == NULL) {
-        URMA_LOG_ERR("data is NULL\n");
-        return -1;
-    }
-
-    bdp_queue_node_t *node = NULL;
-
-    node = STAILQ_FIRST(&q->head);
-    STAILQ_REMOVE_HEAD(&q->head, nodes);
-    *data = node->data;
-    q->node_num--;
-    free(node);
-    return 0;
-}
-
-int bdp_queue_pop_tail(bdp_queue_t *q, void **data)
-{
-    if (STAILQ_EMPTY(&q->head)) {
-        return -1;
-    }
-    if (data == NULL) {
-        URMA_LOG_ERR("data is NULL\n");
-        return -1;
-    }
-
-    bdp_queue_node_t *curr = STAILQ_FIRST(&q->head);
-
-    while (STAILQ_NEXT(curr, nodes)) {
-        curr = STAILQ_NEXT(curr, nodes);
-    }
-    STAILQ_REMOVE(&q->head, curr, bdp_queue_node, nodes);
-    *data = curr->data;
-    q->node_num--;
-    free(curr);
-    return 0;
-}
-
-bool bdp_queue_is_empty(const bdp_queue_t *q)
-{
-    return STAILQ_EMPTY(&q->head);
-}
-
-void bdp_queue_for_each(bdp_queue_t *q, bdp_queue_visit_func_t func)
-{
-    bdp_queue_node_t *node;
-    STAILQ_FOREACH(node, &q->head, nodes) {
-        func(node->data);
-    }
-}
\ No newline at end of file
diff --git a/src/urma/lib/urma/bond/utils/bdp_queue.h b/src/urma/lib/urma/bond/utils/bdp_queue.h
deleted file mode 100644
index 4f5a7a6e..00000000
--- a/src/urma/lib/urma/bond/utils/bdp_queue.h
+++ /dev/null
@@ -1,61 +0,0 @@
-/*
- * SPDX-License-Identifier: MIT
- * Copyright (c) Huawei Technologies Co., Ltd. 2025-2025. All rights reserved.
- * Description: bondp queue header
- * Author: Ma Chuan
- * Create: 2025-03-05
- * Note:
- * History:
- */
-#ifndef BONDP_QUEUE_H
-#define BONDP_QUEUE_H
-
-#include <sys/queue.h>
-#include <stdint.h>
-#include <stdbool.h>
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-#define BDP_QUEUE_ALLOC_ERR (-2)
-
-STAILQ_HEAD(bdp_queue_head, bdp_queue_node);
-
-typedef struct bdp_queue_head bdp_queue_head_t;
-
-typedef struct bdp_queue {
-    bdp_queue_head_t head;
-    uint32_t node_num;
-    uint32_t max_node;
-} bdp_queue_t;
-
-typedef struct bdp_queue_node {
-    void *data;
-    STAILQ_ENTRY(bdp_queue_node) nodes;
-} bdp_queue_node_t;
-
-typedef void (*bdp_queue_visit_func_t)(void *); // input is bdp_queue_node->data
-
-/* User check q is NULL or not */
-
-void bdp_queue_init(bdp_queue_t *q, uint32_t max_node);
-void bdp_queue_uninit(bdp_queue_t *q);
-
-int bdp_queue_front(bdp_queue_t *q, void **data);
-
-int bdp_queue_push_tail(bdp_queue_t *q, void *data);
-
-int bdp_queue_pop_head(bdp_queue_t *q, void **data);
-
-int bdp_queue_pop_tail(bdp_queue_t *q, void **data);
-
-bool bdp_queue_is_empty(const bdp_queue_t *q);
-
-void bdp_queue_for_each(bdp_queue_t *q, bdp_queue_visit_func_t func);
-
-#ifdef __cplusplus
-}
-#endif
-
-#endif // BONDP_QUEUE_H
\ No newline at end of file
-- 
2.28.0.windows.1

